src/test/java/org/jenkinsci/plugins/pagerduty/changeevents/ChangeEventBuilderTest.java

```diff
@@ -0,0 +1,156 @@
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.times;
+
+
+import java.io.IOException;
+
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+
+//power mockito to mock static method of classes
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ChangeEventsAPI.class, DisplayURLProvider.class})
+
+public class ChangeEventBuilderTest {
+	
+	 private static ChangeEventBuilder changeEventBuilder;
+	 private static  AbstractBuild build;
+	 private static  AbstractProject project;
+	 private static  AbstractProject job;
+	 private static  Jenkins jenkins;
+	 private static  Launcher launcher;
+	 private static  BuildListener listener;
+	 private static DisplayURLProvider urlProvider ;
+	 
+	 
+     // setup reusable objects across unit tests inside this method 
+	 @BeforeClass
+	public static void setupMessageBuilder() {
+		 
+		//mock jenkins objects and changeevent builder
+		build = mock(AbstractBuild.class);
+		project = mock(AbstractProject.class);
+		job = mock(AbstractProject.class);
+		jenkins = mock(Jenkins.class);
+		launcher = mock(Launcher.class);
+		listener = mock(BuildListener.class);
+		urlProvider = mock(DisplayURLProvider.class);
+		
+		when(job.getDisplayName()).thenReturn("testjobname");
+		changeEventBuilder = new ChangeEventBuilder("testIntegration key");		
+	 }
+	
+	//Unit test verifying JSON contains the job.getDisplayName() field
+	@Test
+	public void testJobName() throws IOException {
+		ArgumentCaptor<String> jsonArg = ArgumentCaptor.forClass(String.class);
+
+		//static method mocks using PowerMockito
+		PowerMockito.mockStatic(ChangeEventsAPI.class);
+		when(ChangeEventsAPI.send(anyString())).thenReturn(new ChangeEventsAPI.Response(200, "OK"));
+		PowerMockito.mockStatic(DisplayURLProvider.class);
+		when(DisplayURLProvider.get()).thenReturn(urlProvider);
+
+		when(listener.getLogger()).thenReturn(System.out);
+
+		//The field we are testing
+		changeEventBuilder.setSummaryText("testjobname built successfully");
+
+		//trigger build changeevent
+		changeEventBuilder.perform(build, launcher, listener);
+
+		//mock API call
+		PowerMockito.verifyStatic(ChangeEventsAPI.class, times(1));
+		ChangeEventsAPI.send(jsonArg.capture());
+		//Expecting jsonArg to contain a valid JSON string similar to the following
+		//{"payload":{"summary":"testjobname built successfully","source":"Jenkins","custom_details":{"duration":null,"build_number":0},"timestamp":"2021-03-17T21:45:24.808Z"},"links":[{"href":"http://www.testurl.com","text":"View on Jenkins"}],"routing_key":"testIntegration key"}
+
+		assertEquals("testjobname",job.getDisplayName());
+		
+		//verify data passed across
+		assertTrue( jsonArg.getValue().contains("testjobname"));
+		assertTrue( jsonArg.getValue().contains("testjobname built successfully"));
+	}
+
+	//Unit test verifying JSON contains the full URL of the job
```

Comment:
After formatting, I think the two tests testJobName() and testURL() can be combined.  I would like to do ONE test and assert all the things that we want to verify.  I propose testing at minimum these fields:

* summary (default and custom text)
* build_number
* links[0].href
* routing_key